lib/syntax_tree/css/selectors.rb

```diff
@@ -125,7 +125,7 @@ def accept(visitor)
         end
 
         def child_nodes
-          [type, subclasses, pseudo_elements].flatten
+          [type, subclasses, pseudo_elements].compact.flatten
```

Comment:
Up to you, but you could make this:

```suggestion
          [type, *subclasses, *pseudo_elements]
```

Reply:
Unfortunately that change doesn't account for a) `type` being nil and needing to be removed, and b) `pseudo_elements` contains nested arrays.

Reply:
While we're here, I don't really like the fact that pseudo_elements contains nested arrays (see the test "parses a compound selector with pseudo-elements and pseudo-classes" to see what it looks like), but regardless I think `child_nodes` should flatten it all.